### PR TITLE
docs: Add common validation rules requirements

### LINKED ISSUE #17: docs: Specify common rules
# Common validation rules

Define expectations regarding:
- [x] Packaging
- [x] Linting
- [x] Documentation

TLDR: Validation requests must comply with 3 common principles.

## Packaging

- Program must create an executable (an entrypoint script may be used)
- Repository must include some sort of dependency lock file (soft dependencies shall be considered first)

## Linting

- Code must comply with at least one linting community guideline

## Documentation

- Repository must be properly documented
- Installation process must be properly documented for a least a single Linux-based OS (additional OSes may be documented)
- Execution process must be documented (either directly through a helper interface or in the repository documentation)

## Additional common goal

The implementation of a Continuous Integration mechanism is highly encouraged to enforce these expectations.

### PR DESCRIPTION
**Pull request by tbobm**
Closes #17 

image

### CODE REVIEW

File: subjects/rules.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+## Detailed requirements
+
+### Packaging
+
+Code or application **must** be packaged and easily installable.
+
+Any packaging method relevant to the chosen ecosystem can be fit this requirement.
+
+Most common might be Docker
```
Comment (nicolasdecorbez) [inline subjects/rules.md:32]: perhaps, in the future, create an exercise to correctly package applications with docker, for example

**Review by nicolasdecorbez — APPROVED** [review]
No changes required, just a small suggestion for young developers starting out with packaging applications.

**Review by nicolasdecorbez — APPROVED** [review]

File: subjects/rules.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+???+ todo "Validation requests"
+    - program _should_ be able to execute the example(s)
+    - program _should_ accept input given in the subject description excepted
+    if adapting the format is relevant
+    - program _may_ output information in other formats than specified in the subject
+
+???+ todo "Linting"
+    - Repository **must** conform to at least one community linter ([pylint][pylint],
+    [tslint][tslint], `go fmt`).
```
Comment (nicolasdecorbez) [inline subjects/rules.md:28]: You could add [this link](https://blog.golang.org/gofmt) to `go fmt`, if you find it helpful.

**Review by nicolasdecorbez — APPROVED** [review]
Just a brief note about `go fmt`, otherwise I approve theses changes ! 

Thanks for the screenshot too ! Found it more easy to review.

File: subjects/rules.md (hunk 2 above)
Comment (tbobm) [reply subjects/rules.md]: I added a link to go fmt to prevent any misunderstanding, thank ! :rocket:

**Pull request merged**